lib/modules/manager/pub/extract.ts

```diff
@@ -1,64 +1,74 @@
-import { load } from 'js-yaml';
-import { logger } from '../../../logger';
+import is from '@sindresorhus/is';
 import { DartDatasource } from '../../datasource/dart';
+import { FlutterVersionDatasource } from '../../datasource/flutter-version';
 import type { PackageDependency, PackageFileContent } from '../types';
+import type { PubspecSchema } from './schema';
+import { parsePubspec } from './utils';
 
-function getDeps(
-  depsObj: { [x: string]: any },
-  preset: { depType: string }
+function extractFromSection(
+  pubspec: PubspecSchema,
+  sectionKey: keyof Pick<PubspecSchema, 'dependencies' | 'dev_dependencies'>
 ): PackageDependency[] {
-  if (!depsObj) {
+  const sectionContent = pubspec[sectionKey];
+  if (!sectionContent) {
     return [];
   }
-  return Object.keys(depsObj).reduce((acc, depName) => {
+
+  const datasource = DartDatasource.id;
+  const deps: PackageDependency[] = [];
+  for (const depName of Object.keys(sectionContent)) {
     if (depName === 'meta') {
-      return acc;
+      continue;
     }
 
-    const section = depsObj[depName];
-
-    let currentValue: string | null = null;
-    if (section?.version) {
-      currentValue = section.version.toString();
-    } else if (section) {
-      if (typeof section === 'string') {
-        currentValue = section;
-      }
-      if (typeof section === 'number') {
-        currentValue = section.toString();
+    let currentValue = sectionContent[depName];
+    if (!is.string(currentValue)) {
+      const version = currentValue.version;
+      if (version) {
+        currentValue = version;
+      } else {
+        continue;
```

Comment:
set currentValue to undefined and return the dep, renovate will add a skip reason later because of missing value